src/Misc/layoutroot/run.sh

```diff
@@ -9,10 +9,10 @@ while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symli
     [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
 done
 DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
-cp -f "$DIR"/run-helper.sh.template "$DIR"/run-helper.sh
 # run the helper process which keep the listener alive
 while :;
 do
+    cp -f "$DIR"/run-helper.sh.template "$DIR"/run-helper.sh
```

Comment:
This is why the canary tests weren't signalling even though run-helper had the above error: run.sh would use the old version of run-helper.sh until `run.sh` itself was restarted, because the copying of the new `run-helper.sh` was outside the runner restart loop.

When the tests were set up, `run.sh` itself would get restarted once so this went undetected.

This is bash only, the .cmd loop is correct.